typescript/infra/src/utils/violation.ts

```diff
@@ -0,0 +1,206 @@
+      })
+      .map((item) => sortArraysByType(item));
+  } else if (typeof obj === 'object' && obj !== null) {
+    const sortedObj: AnyObject = {};
+    Object.keys(obj).forEach((key) => {
+      sortedObj[key] = sortArraysByType(obj[key]);
+    });
+    return sortedObj;
+  }
+  return obj;
+}
+
+function removeFields(obj: any): any {
+  if (Array.isArray(obj)) {
+    return obj.map(removeFields);
+  } else if (obj !== null && typeof obj === 'object') {
+    return Object.keys(obj).reduce((result: any, key: string) => {
+      if (!ignoreFields.includes(key)) {
+        result[key] = removeFields(obj[key]);
+      }
+      return result;
+    }, {});
+  } else {
+    return obj;
+  }
+}
+
+function logDiff(expected: AnyObject, actual: AnyObject): void {
+  const sortedExpected = sortArraysByType(expected);
+  const sortedActual = sortArraysByType(actual);
+
+  const sortedExpectedJson = stringify(sortedExpected, { space: 2 });
+  const sortedActualJson = stringify(sortedActual, { space: 2 });
+
+  const parsedSortedExpected = JSON.parse(sortedExpectedJson);
+  const parsedSortedActual = JSON.parse(sortedActualJson);
+
+  const added = addedDiff(parsedSortedExpected, parsedSortedActual);
+  const deleted = deletedDiff(parsedSortedExpected, parsedSortedActual);
+  const updated = updatedDiff(parsedSortedExpected, parsedSortedActual);
+
+  const logChanges = (
+    changes: AnyObject,
+    changeType: string,
+    color: (text: string) => string,
+    config: AnyObject,
+  ) => {
+    const logChange = (obj: AnyObject, path: string = '') => {
+      Object.keys(obj).forEach((key) => {
+        const currentPath = path ? `${path}.${key}` : key;
+        if (
+          typeof obj[key] === 'object' &&
+          obj[key] !== null &&
+          !Array.isArray(obj[key])
+        ) {
+          logChange(obj[key], currentPath);
+        } else {
+          console.log(
+            color(
+              `${changeType} ${updatePath(config, currentPath)}: ${stringify(
+                obj[key],
+                {
+                  space: 2,
+                },
+              )}`,
+            ),
+          );
+        }
+      });
+    };
+    logChange(changes);
+  };
+
+  logChanges(added, '+ Added to config:', chalk.green, parsedSortedActual);
+  logChanges(
+    deleted,
+    '- Deleted from config:',
+    chalk.red,
+    parsedSortedExpected,
+  );
+  logChanges(updated, '~ Updated config:', chalk.yellow, parsedSortedExpected);
+}
+
+function preProcessConfig(config: any) {
+  return removeFields(toLowerCaseValues(config));
+}
+
+export function logViolationDetails(violations: CheckerViolation[]): void {
+  for (const violation of violations) {
+    if (violation.type === CoreViolationType.Mailbox) {
+      const mailboxViolation = violation as MailboxViolation;
+
+      console.log(`Mailbox violation ${mailboxViolation.subType} details:`);
+
+      const preProcessedExpectedConfig = preProcessConfig(
+        mailboxViolation.expected,
```

Comment:
@tkporter - `expected` and `actual` on `mailboxViolation` are of type `any`. I think this was done in the sdk to enable flexibility for values across different violations

The functions here are currently written to handle any object. I could look at casting the variable with type `IsmConfig` after validating using the `IsmConfigSchema` and rewrite the functions so they expect IsmConfig. wdyt?

Reply:
I was thinking of not going far out of your way on this